crates/fuzzing/src/oracles.rs

```diff
@@ -479,3 +476,189 @@ pub fn table_ops(config: crate::generators::Config, ops: crate::generators::tabl
         }
     }
 }
+
+/// Do we accept the given wasm module for differential fuzzing between Wasmtime and wasmi? We want
+/// to return early for modules that don't meet our criteria so that the fuzzer learns that these
+/// are not interesting (as they will not lead to additional coverage).
+///
+/// If accepted, returns the name of the memory we will diff and the function we will call.
+fn differential_wasmi_accept_module(wasm: &[u8]) -> Option<(String, String)> {
+    // Find the names of the first memory and first export.
+    let mut memory_name = None;
+    let mut func_name = None;
+    let parser = wasmparser::Parser::new(0);
+    for payload in parser.parse_all(&wasm[..]) {
+        let payload = match payload {
+            Ok(p) => p,
+            Err(_) => return None,
+        };
```

Comment:
This can be tightened up a bit with `let payload = payload.ok()?` and in a few other places too

Reply:
Oh, neat, I didn't know that `Option` was question-mark-operator-friendly too; thanks!